### PR TITLE
minio role 

### PR DESCRIPTION
**Pull request by Bharat-DRS**

### CODE REVIEW

File: vagrant/provisioning/roles/minio/tasks/main.yml [hunk 1]
```diff
@@ -0,0 +1,52 @@
+
+- name: create minio group
+  become: yes
+  group:
+    name: minio
+    gid: 9002
```
Comment (bojmil-armedia) [inline vagrant/provisioning/roles/minio/tasks/main.yml]: No need to define the **gid**. Other groups created don't define it.

File: vagrant/provisioning/roles/minio/tasks/main.yml [hunk 2]
```diff
@@ -0,0 +1,52 @@
...
+    state: present
+- name: create minio user
+  become: yes
+  user:
+    name: minio
+    group: minio
+    shell: /sbin/nologin
```
Comment (bojmil-armedia) [inline vagrant/provisioning/roles/minio/tasks/main.yml:12]: I see in other places we use shell: /bin/false . What is the difference?

File: vagrant/provisioning/roles/minio/tasks/main.yml [hunk 3]
```diff
@@ -0,0 +1,52 @@
...
+    home: /opt/minio
```
Comment (bojmil-armedia) [inline vagrant/provisioning/roles/minio/tasks/main.yml]: If it needs a user home at all, why not /home/minio

File: vagrant/provisioning/roles/minio/tasks/main.yml [hunk 4]
```diff
@@ -0,0 +1,52 @@
...
+- name: create minio directories
+  become: yes
+  file:
+    path: /opt/minio/{{item}}
```
Comment (bojmil-armedia) [inline vagrant/provisioning/roles/minio/tasks/main.yml]: Use {{ root_folder }}/app/minio as base app directory.

File: vagrant/provisioning/roles/minio/tasks/main.yml [hunk 5]
```diff
@@ -0,0 +1,52 @@
...
+  get_url:
+    url: "https://dl.minio.io/server/minio/release/linux-amd64/minio"
+    dest: /opt/minio/bin/
+    mode: '0777'
+- name: create minio config file
+  become: yes
+  copy:
+    dest: /opt/minio/minio.conf
+    content: |
+     MINIO_VOLUMES= /opt/minio/data
```
Comment (bojmil-armedia) [inline vagrant/provisioning/roles/minio/tasks/main.yml]: Use {{ root_folder }}/data/minio as data directory

**Comment by bojmil-armedia** [comment]
Setup logging to {{ root_folder }}/log/minio directory and setup log rotation (see examples in other services).

File: vagrant/provisioning/roles/minio/tasks/main.yml [hunk 6]
```diff
@@ -0,0 +1,52 @@
...
+- name: download minio file from internet
+  become: yes
+  get_url:
+    url: "https://dl.minio.io/server/minio/release/linux-amd64/minio"
+    dest: /opt/minio/bin/
+    mode: '0777'
+- name: create minio config file
+  become: yes
+  copy:
+    dest: /opt/minio/minio.conf
```
Comment (dzdimov) [inline vagrant/provisioning/roles/minio/tasks/main.yml]: Let's use {{ root_folder }} variable here so instead /opt/minio/minio.conf we should use {{ root_folder }}/app/minio/minio.conf

We use {{ root_folder }}/app for application location and {{ root_folder }}/data for data the service is storing and {{ root_folder }}/log for the log location

File: vagrant/provisioning/roles/minio/tasks/main.yml [hunk 7]
```diff
@@ -0,0 +1,52 @@
...
+  become: yes
+  template:
+    src: "minio.service"
+    dest: /etc/systemd/system/minio.service
+- name: enable minio service
+  become: yes
+  systemd:
+    name: minio
+    enabled: yes
+    masked: no
```
Comment (dzdimov) [inline vagrant/provisioning/roles/minio/tasks/main.yml:45]: do we need daemon-reload?

File: vagrant/provisioning/roles/minio/templates/minio.service [hunk 8]
```diff
@@ -0,0 +1,38 @@
+[Unit]
+Description=Minio
+Documentation=https://docs.minio.io
+Wants=network-online.target
+After=network-online.target
+AssertFileIsExecutable=/opt/minio/bin/minio
+
+[Service]
+WorkingDirectory=/opt/minio
```
Comment (dzdimov) [inline vagrant/provisioning/roles/minio/templates/minio.service]: make sure this reflects the {{ root_folder }} location

File: vagrant/provisioning/roles/minio/templates/minio.service [hunk 9]
```diff
@@ -0,0 +1,38 @@
...
+
+User=minio
+Group=minio
+
+PermissionsStartOnly=true
+
+EnvironmentFile=-/opt/minio/minio.conf
```
Comment (dzdimov) [inline vagrant/provisioning/roles/minio/templates/minio.service]: make sure this reflects the {{ root_folder }} location

File: vagrant/provisioning/roles/minio/templates/minio.service [hunk 10]
```diff
@@ -0,0 +1,38 @@
...
+ExecStartPre=/bin/bash -c "[ -n \"${MINIO_VOLUMES}\" ] || echo \"Variable MINIO_VOLUMES not set in /opt/minio/minio.conf\""
```
Comment (dzdimov) [inline vagrant/provisioning/roles/minio/templates/minio.service:17]: make sure this reflects the {{ root_folder }} location

File: vagrant/provisioning/roles/minio/templates/minio.service [hunk 11]
```diff
@@ -0,0 +1,38 @@
...
+
+ExecStart=/opt/minio/bin/minio server $MINIO_OPTS $MINIO_VOLUMES
```
Comment (dzdimov) [inline vagrant/provisioning/roles/minio/templates/minio.service:19]: make sure this reflects the {{ root_folder }} location

**Comment by jovestefanovski** [comment]
Add the role in arkcase-dev.yml

**Comment by jovestefanovski** [comment]
create firewall rule to be able to access minio console

File: vagrant/provisioning/roles/minio/tasks/main.yml (hunk 2 above)
Comment (david-oc-miller) [reply vagrant/provisioning/roles/minio/tasks/main.yml:12]: it seems with /sbin/nologin the shell prints a message "this account is not available" but with /bin/false the shell just terminates with no message.  Not much difference but we should use /sbin/nologin everywhere.  https://unix.stackexchange.com/questions/10852/whats-the-difference-between-sbin-nologin-and-bin-false

File: vagrant/provisioning/roles/minio/tasks/main.yml (hunk 6 above)
Comment (david-oc-miller) [reply vagrant/provisioning/roles/minio/tasks/main.yml]: {{ root_folder }}/tmp for temp files, if MinIO uses them, and {{ root_folder }}/install to download install artifacts into...

File: vagrant/provisioning/roles/minio/tasks/main.yml (hunk 3 above)
Comment (david-oc-miller) [reply vagrant/provisioning/roles/minio/tasks/main.yml]: We don't have a {{ root_folder }}/home concept, so the normal /home is better.

File: vagrant/provisioning/roles/minio/tasks/main.yml [hunk 12]
```diff
@@ -0,0 +1,52 @@
...
+    owner: minio
+  with_items:
+    - 'bin'
+    - 'data'
+- name: download minio file from internet
+  become: yes
+  get_url:
+    url: "https://dl.minio.io/server/minio/release/linux-amd64/minio"
+    dest: /opt/minio/bin/
+    mode: '0777'
```
Comment (david-oc-miller) [inline vagrant/provisioning/roles/minio/tasks/main.yml:28]: {{ root_folder }}/install/minio to download install artifacts into.

**Comment by bojmil-armedia** [comment]
> Add the role in arkcase-dev.yml

Also put it in arkcase-all.yml with following tags:
    - role: minio
      tags: [microservices, minio]

File: vagrant/provisioning/roles/minio/tasks/main.yml (hunk 7 above)
Comment (david-oc-miller) [reply vagrant/provisioning/roles/minio/tasks/main.yml:45]: The Ansible `systemd` module has a `daemon-reload` attribute, if set to true we shouldn't need another command task to call it through the shell: https://docs.ansible.com/ansible/latest/collections/ansible/builtin/systemd_module.html#parameter-daemon_reload

File: vagrant/provisioning/roles/minio/tasks/main.yml [hunk 13]
```diff
@@ -3,19 +3,18 @@
...
     name: minio
     group: minio
     shell: /sbin/nologin
-    home: /opt/minio
+    home: /home/minio
 - name: create minio directories
   become: yes
   file:
-    path: /opt/minio/{{item}}
+    path: "{{root_folder}}/minio/{{item}}"
```
Comment (david-oc-miller) [inline vagrant/provisioning/roles/minio/tasks/main.yml:17]: We want a folder structure like so:

{{ root_folder }}/app/minio

{{ root_folder }}/log/minio

{{ root_folder }}/data/minio

{{ root_folder }}/tmp/minio

{{ root_folder }}/install/minio

**Comment by david-oc-miller** [comment]
@Bharat-DRS do we still need this merge request, or is it replaced by #395 ?

**Comment by Bharat-DRS** [comment]
As PR:395 is already merged to Develop branch we can close this duplicate PR

**Pull request closed without merging**